apps/services/notifications/src/app/modules/notifications/queueConnection.provider.ts

```diff
@@ -0,0 +1,73 @@
+import assert from 'assert'
+import { Inject, Injectable, OnApplicationBootstrap } from '@nestjs/common'
+import {
+  SQSClient,
+  CreateQueueCommand,
+  GetQueueAttributesCommand,
+} from '@aws-sdk/client-sqs'
```

Comment:
Can you use/extend `libs/message-queue` in this service?

The reason it exists is that we wanted to abstract proprietary SQS code in one place so it is easier to replace if we have to.

Reply:
We probably could. I didn't use it because it was undocumented, uses AWS SNS which I'm not sure why we need, and doesn't support graceful shutdown of workers.

The documentation and graceful shutdown is something I could fix.
Can you explain to me why it uses SNS in stead of pushing directly to the queue?

Reply:
I can give some context into that. We used SNS to fanout messages to multiple SQS queues.
So using SNS in front of SQS gives you the option to send a single message to SNS and SNS will send that message to all the connected SQS queues.

Reply:
It's a little bit more boilerplate creating the exhange(SNS) and then binding it to the queue but it also gives us the possibility to do more advanced routing of our messages.